tests: bluetooth: classic: Add test suite a2dp source

CC0918 opened the pull request:
IUT works as a a2dp source. The peer device, a2dp sink is a PC with running bumble on it.

The test content are follows:
case 1: test API register_ep. The api is used to register endpoint. The case tests it when the parameters is correct. case 2: test API a2dp connect. This case repeats the a2dp connect operation to verify its correctness
case 3: test API a2dp disconnect.This case repeats the a2dp connect operation to verify its correctness
case 4: test API a2dp discover. The case send discover and get (all)capabilities cmd to a2dp source.
case 5: This case will complete the execution process of the test a2dp, including connect,discover,get cap, open, start,suspend,close. All cmd are send from sink to source.
case 6: This case will complete the execution process of the test a2dp, including connect,discover,get cap, open, start,suspend,close. All cmd are send from source to sink.

lylezhu2012 commented on tests/bluetooth/classic/a2dp_source/src/test_a2dp_source.c:
My suggestion is,
```
	if (BT_A2DP_SBC != codec_type) {
		bt_shell_print("  not SBC codecs");
		return;
	}

	...
```

lylezhu2012 commented on tests/bluetooth/classic/a2dp_source/src/test_a2dp_source.c:
```diff
@@ -0,0 +1,815 @@
+	bt_shell_print("endpoint id: %d, %s, %s:", ep_info->sep_info->id,
+		       (ep_info->sep_info->tsep == BT_AVDTP_SINK) ? "(sink)" : "(source)",
+		       (ep_info->sep_info->inuse) ? "(in use)" : "(idle)");
+	if (BT_A2DP_SBC == codec_type) {
```
My suggestion is,
```
	if (BT_A2DP_SBC != codec_type) {
		bt_shell_print("  not SBC codecs");
		return;
	}

	...
```

lylezhu2012 commented on tests/bluetooth/classic/a2dp_source/src/test_a2dp_source.c:
```diff
@@ -0,0 +1,815 @@
+#endif
+	} else {
+		bt_shell_print("a2dp connecting fail");
+	}
```
``` suggestion
	if (err != 0) {
		bt_shell_print("a2dp connecting fail");
		return;
	}

	default_a2dp = a2dp;
	bt_shell_print("a2dp connected");
	if (role == BT_AVDTP_SOURCE) {
		k_work_init_delayable(&send_media, a2dp_send_media_timeout);
	}
```

lylezhu2012 commented on tests/bluetooth/classic/a2dp_source/src/test_a2dp_source.c:
```diff
@@ -0,0 +1,815 @@
+static void app_start_rsp(struct bt_a2dp_stream *stream, uint8_t rsp_err_code)
+{
+	if (rsp_err_code == 0) {
+		bt_shell_print("success to start");
```
Shall the delayable worker `send_media` be scheduled after the `start` response has been received without any errors?

lylezhu2012 commented on tests/bluetooth/classic/a2dp_source/src/test_a2dp_source.c:
```diff
@@ -0,0 +1,815 @@
+
+static int cmd_register_ep(const struct shell *sh, int32_t argc, char *argv[])
+{
+	int err = -1;
```
Hard error code `-1` should not be used here.

lylezhu2012 commented on tests/bluetooth/classic/a2dp_source/src/test_a2dp_source.c:
```diff
@@ -0,0 +1,815 @@
+		shell_print(sh, "need to register a2dp connection callbacks");
+		return -ENOEXEC;
+	}
+
```
Duplicated code block.

CC0918 committed:
tests: bluetooth: classic: Add test suite a2dp source

IUT works as a a2dp source. The peer device, a2dp sink is
a PC with running bumble on it.

The test content are follows:
case 1: test API register_ep. The api is used to register endpoint.
The case tests it when the parameters is correct or incorrect.
case 2: test API a2dp connect. This case repeats the a2dp connect
operation to verify its correctness
case 3: test API a2dp disconnect.This case repeats the a2dp connect
operation to verify its correctness
case 4: test API a2dp discover. The case send discover and get
(all)capabilities cmd to a2dp source.
case 5: This case will complete the execution process of the test a2dp,
including connect,discover,get cap, open, start,suspend,close.
All cmd are send from sink to source.
case 6: This case will complete the execution process of the test a2dp,
including connect,discover,get cap, open, start,suspend,close.
All cmd are send from source to sink.

Signed-off-by: Cheng Chang <cheng.chang@nxp.com>

chengkai15 commented on tests/bluetooth/classic/a2dp_source/pytest/test_a2dp.py:
it seems that it was defined but never called anywhere in the test suite

suggest  remove generate_packets() or add a test case that 
uses it for media packet verification. 

dead code in tests is confusing

chengkai15 replied:
sorry copilot has catch it
